force-app/main/default/classes/GiftBatch.cls

```diff
@@ -122,6 +130,32 @@ public inherited sharing class GiftBatch implements IGiftBatch {
         this.totals = new GiftBatchTotals(giftBatchId, giftBatchSelector);
     }
 
+    public Boolean shouldPayFirstInstallment () {
+        return this.shouldPayFirstInstallment;
+    }
+
+    private Boolean calculateShouldPayFirstInstallment() {
+        JSONParser parser = JSON.createParser(this.batch.Batch_Defaults__c);
+        String shouldPayFirstInstallment;
+
+        while (parser.nextToken() != null) {
+            if (parser.getCurrentToken() == JSONToken.FIELD_NAME) {
+                if (parser.getText() == AllowFirstInstallmentPaid) {
+                    parser.nextToken();
```

Comment:
Gonna need an explanation for this one, how do we know how many times to call it?

Reply:
The string being parsed is `{"AllowFirstInstallment__f":{"objectApiName":"AllowFirstInstallment__o","fieldApiName":"AllowFirstInstallment__f","value":true}`. This structure is never going to change since we defined it in the front-end. The amount of times we call `parser.nextToken()` corresponds to the number of tokens that exist before the `value` token that holds the boolean we want.

Reply:
I know we try to avoid comments and let the code speak for itself, but I think this needs one, even just a line explaining the approach.

Reply:
Should this have an inner loop that goes through the tokens until it hits the 'value' token.  That way its not as hardcoded to calling a specific number of times?